tf_agents/agents/dqn/dqn_agent.py

```diff
@@ -364,8 +349,10 @@ def _train(self, experience, weights):
       eager_utils.add_gradients_summaries(grads_and_vars,
                                           self.train_step_counter)
     training.apply_gradients(self._optimizer,
-                             grads_and_vars,
```

Comment:
replace all this with
```

self._optimizer.apply_gradients(grads_and_vars)

# Increment train_step_counter
self.train_step_counter.assign_add(1)
```

Reply:
Apologies for the delay. I just hadn't seen your new replacement request. Thank you.